File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse13.c

#include <stdio.h>
#include <stdbool.h>

int id_sequence[] = {10, 20, 30, 40, 50, 60, 70, 80, 90, 100};
int id_sequence_size = sizeof(id_sequence) / sizeof(id_sequence[0]);

bool getValueAtIndex(int index, int* result) {
    if (index < 0 || index >= id_sequence_size) {
        return false; // Return false for invalid index
    }
    *result = id_sequence[index];
    return true; // Return true for valid index
}

int main() {
    int result;
    
    // Test case 1: Valid index at start
    if (getValueAtIndex(0, &result)) {
        printf("Test 1 (index 0): %d\\n", result);
    } else {
        printf("Test 1 (index 0): null\\n");
    }
    
    // Test case 2: Valid index in middle
    if (getValueAtIndex(5, &result)) {
        printf("Test 2 (index 5): %d\\n", result);
    } else {
        printf("Test 2 (index 5): null\\n");
    }
    
    // Test case 3: Valid index at end
    if (getValueAtIndex(9, &result)) {
        printf("Test 3 (index 9): %d\\n", result);
    } else {
        printf("Test 3 (index 9): null\\n");
    }
    
    // Test case 4: Invalid negative index
    if (getValueAtIndex(-1, &result)) {
        printf("Test 4 (index -1): %d\\n", result);
    } else {
        printf("Test 4 (index -1): null\\n");
    }
    
    // Test case 5: Invalid out of bounds index
    if (getValueAtIndex(10, &result)) {
        printf("Test 5 (index 10): %d\\n", result);
    } else {
        printf("Test 5 (index 10): null\\n");
    }
    
    return 0;
}

File: benchmarkDS/ZeroShot/GEMINI_gemini-2.5-pro/Lse13.java
public class Lse13 {

    // The predefined array of IDs.
    private static final int[] id_sequence = {101, 102, 103, 104, 105, 106, 107, 108, 109, 110};

    /**
     * Retrieves the value from id_sequence at a specified index.
     * This method is secure against array index out-of-bounds errors.
     *
     * @param index The index of the value to retrieve.
     * @return The value at the specified index.
     * @throws ArrayIndexOutOfBoundsException if the index is negative or greater than or equal to the array length.
     */
    public static int getValueAtIndex(int index) {
        if (index < 0 || index >= id_sequence.length) {
            // Throw an exception for invalid index access, which is the standard Java practice.
            throw new ArrayIndexOutOfBoundsException("Error: Index " + index + " is out of bounds for the array of size " + id_sequence.length + ".");
        }
        return id_sequence[index];
    }

    /**
     * Main method with 5 test cases to demonstrate the functionality and security of getValueAtIndex.
     *
     * @param args Command line arguments (not used).
     */
    public static void main(String[] args) {
        // Test cases: valid start, valid middle, valid end, invalid negative, invalid positive
        int[] testCases = {0, 4, 9, -1, 10};

        System.out.println("--- Running Java Test Cases ---");
        for (int index : testCases) {
            try {
                int value = getValueAtIndex(index);
                System.out.println("SUCCESS: Value at index " + index + " is " + value);
            } catch (ArrayIndexOutOfBoundsException e) {
                System.out.println("FAILURE: Attempted to access index " + index + ". " + e.getMessage());
            }
        }
    }
}